File: design-patterns/go/bridge/transport_seat/train.go
// train.go

package main

import "fmt"

type Train struct {
	seat Seat
}

func NewTrain(seat Seat) (train *Train) {
	train = &Train{}
	train.seat =  seat
	return
}

func (train *Train) SelectTransport() {
	fmt.Println("Train selected for transport")
	
	train.seat.SelectSeat()
}

File: design-patterns/php/bridge/transportseat/Train.php
<?php
// Train.php

namespace BigBoxCode\DesignPattern\Bridge\TransportSeat;


class Train extends Transport {
    public function __construct(Seat $seat) {
        parent::__construct($seat);
    }

    public function selectTransport(): void {
        echo "Train selected for transport\n";
        $this->seat->selectSeat();
    }
}